pubsub/index.js

```diff
@@ -6,239 +6,172 @@
+        return outer_event.user_id === inner_event.user_id
+      })
+    })
+
+    if (identify_events.length > 0) {
+      let identifying_response = await retry (async bail => {
+        let identifying_response = await request('https://api.amplitude.com/identify', {
+          method: 'POST',
+          lookup,
+          formData: {
+            api_key: AMPLITUDE_API_KEY,
+            identification: JSON.stringify(identify_events)
+          },
+          timeout: 5 * 1000
+        })
+
+        return identifying_response
+      }, {
+        minTimeout: 15000,
+        maxTimeout: 60000,
+        onRetry: error => {
+          logger.error({type: 'identify.error', error: error})
+        }
+      })
+      
+      if (identifying_response == 'success') {
+        logger.info({type: 'identify.success', count: identify_events.length}, 'Identify success')
+      } else {
+        throw new Error('Something goofed in the Identify API')
       }
     }
-  }
-
-  if (isIgnoredEvent(event)) {
-    return {}
-  }
-
-  if (! isEventOk(event)) {
-    console.log(timestamp(), 'Warning: Skipping malformed event', event)
-    return {}
-  }
-
-  if (event.user_id) {
-    event.user_id = hash(event.user_id)
-  }
 
-  event.insert_id = hash(event.user_id, event.device_id, event.session_id, event.event_type, event.time)
-
-  let identify
-  if (IDENTIFY_VERBS.some(verb => is.assigned(event.user_properties[verb]))) {
-    identify = {
-      device_id: event.device_id,
-      user_id: event.user_id,
-      user_properties: splitIdentifyPayload(event.user_properties),
-      // _insert_id is only here so we can uniquely identify each payload and
-      // link it back to its message. It's not actually sent to Amplitude.
-      _insert_id: event.insert_id,
+    let http_response = await retry (async bail => {
+      let http_response = await request('https://api.amplitude.com/2/httpapi', {
+        method: 'POST',
+        lookup,
+        json: true,
+        body: {
+          api_key: AMPLITUDE_API_KEY,
+          events: http_events
+        },
+        timeout: 5 * 1000
+      }, {
+        onRetry: error => {
+          logger.error({type: 'http.error', error: error})
+        }
+      })
+
+      return http_response
+    })
+
+    if (http_response.code == 200) {
+      logger.info({type: 'http.success', count: http_events.length}, 'HTTP success')
+    } else {
+      logger.info(http_response)
+      throw new Error('Something goofed in the HTTP API')
     }
-  }
-
-  return {
-    httpapi: event,
-    identify,
-  }
-}
-
-function isIgnoredEvent (event) {
-  if (! IGNORED_EVENTS.has(event.event_type)) {
-    return false
-  }
-
-  const criteria = IGNORED_EVENTS.get(event.event_type)
 
-  if (Array.isArray(criteria)) {
-    return criteria.some(criterion => deepMatch(event, criterion))
-  }
+    payload.forEach(message => message.ack())
+  }, MAX_EVENTS_PER_BATCH)
 
-  return deepMatch(event, criteria)
+  setTimeout(() => {
```

Comment:
Do we ever need to clear this timeout, e.g., if all the events have been processed? In the original version the script quit if no new messages came in after some threshold, is that no longer needed?